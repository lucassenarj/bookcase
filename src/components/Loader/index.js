import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./style.css";

function Loader({ loading }) {
  return (
    <>
     {
       loading && (
        <div className="preloader">
          <div className="loader">
            <div className="ytp-spinner">
              <div className="ytp-spinner-container">
                <div className="ytp-spinner-rotator">
                  <div className="ytp-spinner-left">
                    <div className="ytp-spinner-circle"></div>
                  </div>
                  <div className="ytp-spinner-right">
                    <div className="ytp-spinner-circle"></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
       )
     }
    </>
  );
}

Loader.propTypes = {
  loading: PropTypes.bool.isRequired,
}

function mapStateToProps(state) {
  return {
    loading: state.app.loading,
  };
}

export default connect(mapStateToProps)(Loader);
