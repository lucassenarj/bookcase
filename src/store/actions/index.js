export * from "./app";
export * from "./books";