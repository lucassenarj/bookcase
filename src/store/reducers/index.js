import { combineReducers } from "redux";

import app from "./app";
import books from "./books";

const rootReducer = combineReducers({
  app,
  books,
});

export default rootReducer;
