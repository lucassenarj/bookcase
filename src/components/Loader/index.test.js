import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from 'react-redux';
import { render } from "@testing-library/react";
import store from "store";
import Loader from "./";

const mockStore = configureStore([]);

describe("Loader Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Loader />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("loading not showing", () => {
    const mockedStore = mockStore({
      app: {
        loading: false,
      },
    });
  
    const { container } = render(
      <Provider store={mockedStore}>
        <Loader />
      </Provider>
    );
  
    expect(container.firstChild).toBeNull();
  });
});