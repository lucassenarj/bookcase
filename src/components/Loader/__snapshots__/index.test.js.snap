// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders correctly 1`] = `
<div
  class="preloader"
>
  <div
    class="loader"
  >
    <div
      class="ytp-spinner"
    >
      <div
        class="ytp-spinner-container"
      >
        <div
          class="ytp-spinner-rotator"
        >
          <div
            class="ytp-spinner-left"
          >
            <div
              class="ytp-spinner-circle"
            />
          </div>
          <div
            class="ytp-spinner-right"
          >
            <div
              class="ytp-spinner-circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
